cloud/shared/bin/lib/setup.py

```diff
@@ -0,0 +1,31 @@
+import argparse
+import subprocess
+
+from config_validator import ConfigValidator
+from write_tfvars import TfVarWritter
+
+parser = argparse.ArgumentParser()
+
+parser.addArgument("-c", "--config_filename", type=string, help="the filename for the config file")
+parser.addArgument("-v", "--variable_definition_filename", type=string, help="filename for variable definition")
+
+validator = ConfigValidator(parser.config_filename, parser.variable_definition_filename)
+is_valid, validation_errors = validator.validate_config()
+
+if not is_valid:
+    exit(f'Found the following validation errors {validation_errors.join(' ')}')
+
+variables = validator.get_config_variables()
```

Comment:
this looks like more than validation since it's also presenting the vars. maybe `ConfigLoader`?

Reply:
or could factor out a `ConfigLoader` class that `ConfigValidator` can call methods on

Reply:
ooo yes this is great i was struggling to think of a name